### ISSUE TITLE
Overtemp shutdowns on Dell XPS 13 9365

### ISSUE THREAD

**Issue by tbaumann** [intel/thermal_daemon#325]
Hi,

I quite often get thermal protection shutdown events. (Logged in BIOS as such)

I think the issue is that the warning and critical threshold are too close to each other. It seems like the system does a forced shutdown when it reaches 100 C.

Here is the startup logs from thermald. Version 2.4.6 from Ubuntu 21.10 verbose mode
[thermald.service.startup.log](https://github.com/intel/thermal_daemon/files/7563199/thermald.service.startup.log)

Those are the logs leading up to previous forced shutdown

```

Nov 18 13:23:17 zuse-klappi thermald[1669]: set cdev state index 13 state 8500000 wr:8500000
Nov 18 13:23:17 zuse-klappi thermald[1669]: Set : 96000, 93050, 13, 8500000, 2500000
Nov 18 13:23:38 zuse-klappi thermald[1669]: Added zone 9 trip 0 clamp_valid 0 clamp 9000000
Nov 18 13:23:38 zuse-klappi thermald[1669]: set cdev state index 13 state 8800000 wr:8800000
Nov 18 13:23:38 zuse-klappi thermald[1669]: Set : threshold:96000, temperature:96050, cdev:13(rapl_controller_mmio), curr_state:8800000, max_state:2500000
Nov 18 13:23:40 zuse-klappi thermald[1669]: Erased  [9: 0 9000000
Nov 18 13:23:40 zuse-klappi thermald[1669]: set cdev state index 13 state 8500000 wr:8500000
Nov 18 13:23:40 zuse-klappi thermald[1669]: Set : 96000, 89050, 13, 8500000, 2500000
Nov 18 13:23:57 zuse-klappi thermald[1669]: Added zone 9 trip 0 clamp_valid 0 clamp 9000000
Nov 18 13:23:57 zuse-klappi thermald[1669]: set cdev state index 13 state 8800000 wr:8800000
Nov 18 13:23:57 zuse-klappi thermald[1669]: Set : threshold:96000, temperature:98050, cdev:13(rapl_controller_mmio), curr_state:8800000, max_state:2500000
Nov 18 13:23:58 zuse-klappi thermald[1669]: Erased  [9: 0 9000000
Nov 18 13:23:58 zuse-klappi thermald[1669]: set cdev state index 13 state 8500000 wr:8500000
Nov 18 13:23:58 zuse-klappi thermald[1669]: Set : 96000, 95050, 13, 8500000, 2500000
Nov 18 13:24:54 zuse-klappi thermald[1669]: Added zone 9 trip 0 clamp_valid 0 clamp 9000000
Nov 18 13:24:54 zuse-klappi thermald[1669]: set cdev state index 13 state 6218000 wr:6218000
Nov 18 13:24:54 zuse-klappi thermald[1669]: Set : threshold:96000, temperature:97050, cdev:13(rapl_controller_mmio), curr_state:6218000, max_state:2500000
Nov 18 13:24:56 zuse-klappi thermald[1669]: Erased  [9: 0 9000000
Nov 18 13:24:56 zuse-klappi thermald[1669]: set cdev state index 13 state 8500000 wr:8500000
Nov 18 13:24:56 zuse-klappi thermald[1669]: Set : 96000, 86050, 13, 8500000, 2500000
Nov 18 13:25:06 zuse-klappi thermald[1669]: Added zone 9 trip 0 clamp_valid 0 clamp 9000000
Nov 18 13:25:06 zuse-klappi thermald[1669]: set cdev state index 13 state 8800000 wr:8800000
Nov 18 13:25:06 zuse-klappi thermald[1669]: Set : threshold:96000, temperature:99050, cdev:13(rapl_controller_mmio), curr_state:8800000, max_state:2500000
Nov 18 13:25:07 zuse-klappi thermald[1669]: Erased  [9: 0 9000000
Nov 18 13:25:07 zuse-klappi thermald[1669]: set cdev state index 13 state 8500000 wr:8500000
Nov 18 13:25:07 zuse-klappi thermald[1669]: Set : 96000, 90050, 13, 8500000, 2500000
Nov 18 13:25:42 zuse-klappi thermald[1669]: Added zone 9 trip 0 clamp_valid 0 clamp 9000000
Nov 18 13:25:42 zuse-klappi thermald[1669]: set cdev state index 13 state 8800000 wr:8800000
Nov 18 13:25:42 zuse-klappi thermald[1669]: Set : threshold:96000, temperature:98050, cdev:13(rapl_controller_mmio), curr_state:8800000, max_state:2500000
Nov 18 13:25:45 zuse-klappi thermald[1669]: Erased  [9: 0 9000000
Nov 18 13:25:45 zuse-klappi thermald[1669]: set cdev state index 13 state 8500000 wr:8500000
Nov 18 13:25:45 zuse-klappi thermald[1669]: Set : 96000, 88050, 13, 8500000, 2500000
Nov 18 13:26:39 zuse-klappi thermald[1669]: Added zone 9 trip 0 clamp_valid 0 clamp 9000000
Nov 18 13:26:39 zuse-klappi thermald[1669]: set cdev state index 13 state 8800000 wr:8800000
Nov 18 13:26:39 zuse-klappi thermald[1669]: Set : threshold:96000, temperature:96050, cdev:13(rapl_controller_mmio), curr_state:8800000, max_state:2500000
Nov 18 13:26:41 zuse-klappi thermald[1669]: Erased  [9: 0 9000000
Nov 18 13:26:41 zuse-klappi thermald[1669]: set cdev state index 13 state 8500000 wr:8500000
Nov 18 13:26:41 zuse-klappi thermald[1669]: Set : 96000, 87050, 13, 8500000, 2500000
Nov 18 13:26:54 zuse-klappi thermald[1669]: Added zone 9 trip 0 clamp_valid 0 clamp 9000000
Nov 18 13:26:54 zuse-klappi thermald[1669]: set cdev state index 13 state 8800000 wr:8800000
Nov 18 13:26:54 zuse-klappi thermald[1669]: Set : threshold:96000, temperature:96050, cdev:13(rapl_controller_mmio), curr_state:8800000, max_state:2500000
Nov 18 13:26:57 zuse-klappi thermald[1669]: Erased  [9: 0 9000000
Nov 18 13:26:57 zuse-klappi thermald[1669]: set cdev state index 13 state 8500000 wr:8500000
Nov 18 13:26:57 zuse-klappi thermald[1669]: Set : 96000, 89050, 13, 8500000, 2500000
Nov 18 13:27:08 zuse-klappi thermald[1669]: Added zone 9 trip 0 clamp_valid 0 clamp 9000000
Nov 18 13:27:08 zuse-klappi thermald[1669]: set cdev state index 13 state 8800000 wr:8800000
Nov 18 13:27:08 zuse-klappi thermald[1669]: Set : threshold:96000, temperature:99050, cdev:13(rapl_controller_mmio), curr_state:8800000, max_state:2500000
Nov 18 13:27:10 zuse-klappi thermald[1669]: Erased  [9: 0 9000000
Nov 18 13:27:10 zuse-klappi thermald[1669]: set cdev state index 13 state 8500000 wr:8500000
Nov 18 13:27:10 zuse-klappi thermald[1669]: Set : 96000, 94050, 13, 8500000, 2500000
Nov 18 13:27:32 zuse-klappi thermald[1669]: Added zone 9 trip 0 clamp_valid 0 clamp 9000000
Nov 18 13:27:32 zuse-klappi thermald[1669]: set cdev state index 13 state 8800000 wr:8800000
Nov 18 13:27:32 zuse-klappi thermald[1669]: Set : threshold:96000, temperature:96050, cdev:13(rapl_controller_mmio), curr_state:8800000, max_state:2500000
Nov 18 13:27:34 zuse-klappi thermald[1669]: Erased  [9: 0 9000000
Nov 18 13:27:34 zuse-klappi thermald[1669]: set cdev state index 13 state 8500000 wr:8500000
Nov 18 13:27:34 zuse-klappi thermald[1669]: Set : 96000, 92050, 13, 8500000, 2500000
Nov 18 13:27:45 zuse-klappi thermald[1669]: Added zone 9 trip 0 clamp_valid 0 clamp 9000000
Nov 18 13:27:45 zuse-klappi thermald[1669]: cdev index:13 consecutive call, increment exponentially state 8600000 (min 9000000 max 2500000)
Nov 18 13:27:45 zuse-klappi thermald[1669]: set cdev state index 13 state 8600000 wr:8600000
Nov 18 13:27:45 zuse-klappi thermald[1669]: Set : threshold:96000, temperature:98050, cdev:13(rapl_controller_mmio), curr_state:8600000, max_state:2500000
Nov 18 13:27:48 zuse-klappi thermald[1669]: Erased  [9: 0 9000000
Nov 18 13:27:48 zuse-klappi thermald[1669]: set cdev state index 13 state 8500000 wr:8500000
Nov 18 13:27:48 zuse-klappi thermald[1669]: Set : 96000, 93050, 13, 8500000, 2500000
Nov 18 13:28:00 zuse-klappi thermald[1669]: Added zone 5 trip 2 clamp_valid 1 clamp 4500000
Nov 18 13:28:00 zuse-klappi thermald[1669]: set cdev state index 13 state 4500000 wr:4500000
Nov 18 13:28:00 zuse-klappi thermald[1669]: Set : 56000, 56050, 13, 4500000, 2500000
Nov 18 13:28:01 zuse-klappi thermald[1669]: Erased  [5: 2 4500000
Nov 18 13:28:01 zuse-klappi thermald[1669]: set cdev state index 13 state 8500000 wr:8500000
Nov 18 13:28:01 zuse-klappi thermald[1669]: Set : 56000, 55050, 13, 8500000, 2500000
Nov 18 13:28:10 zuse-klappi thermald[1669]: Added zone 5 trip 2 clamp_valid 1 clamp 4500000
Nov 18 13:28:10 zuse-klappi thermald[1669]: set cdev state index 13 state 4500000 wr:4500000
Nov 18 13:28:10 zuse-klappi thermald[1669]: Set : 56000, 56050, 13, 4500000, 2500000
Nov 18 13:28:11 zuse-klappi thermald[1669]: Erased  [5: 2 4500000
Nov 18 13:28:11 zuse-klappi thermald[1669]: set cdev state index 13 state 8500000 wr:8500000
Nov 18 13:28:11 zuse-klappi thermald[1669]: Set : 56000, 55050, 13, 8500000, 2500000
Nov 18 13:28:20 zuse-klappi thermald[1669]: Added zone 5 trip 2 clamp_valid 1 clamp 4500000
Nov 18 13:28:20 zuse-klappi thermald[1669]: set cdev state index 13 state 4500000 wr:4500000
Nov 18 13:28:20 zuse-klappi thermald[1669]: Set : 56000, 56050, 13, 4500000, 2500000
Nov 18 13:28:36 zuse-klappi thermald[1669]: Erased  [5: 2 4500000
Nov 18 13:28:36 zuse-klappi thermald[1669]: set cdev state index 13 state 8500000 wr:8500000
Nov 18 13:28:36 zuse-klappi thermald[1669]: Set : 56000, 55050, 13, 8500000, 2500000
Nov 18 13:30:05 zuse-klappi thermald[1669]: Added zone 5 trip 2 clamp_valid 1 clamp 4500000
Nov 18 13:30:05 zuse-klappi thermald[1669]: set cdev state index 13 state 4500000 wr:4500000
Nov 18 13:30:05 zuse-klappi thermald[1669]: Set : 56000, 56050, 13, 4500000, 2500000
Nov 18 13:30:06 zuse-klappi thermald[1669]: Erased  [5: 2 4500000
Nov 18 13:30:06 zuse-klappi thermald[1669]: set cdev state index 13 state 8500000 wr:8500000
Nov 18 13:30:06 zuse-klappi thermald[1669]: Set : 56000, 55050, 13, 8500000, 2500000
Nov 18 13:30:13 zuse-klappi thermald[1669]: Added zone 9 trip 0 clamp_valid 0 clamp 9000000
Nov 18 13:30:13 zuse-klappi thermald[1669]: set cdev state index 13 state 7850000 wr:7850000
Nov 18 13:30:13 zuse-klappi thermald[1669]: Set : threshold:96000, temperature:96050, cdev:13(rapl_controller_mmio), curr_state:7850000, max_state:2500000
Nov 18 13:30:17 zuse-klappi thermald[1669]: Erased  [9: 0 9000000
Nov 18 13:30:17 zuse-klappi thermald[1669]: set cdev state index 13 state 8500000 wr:8500000
Nov 18 13:30:17 zuse-klappi thermald[1669]: Set : 96000, 90050, 13, 8500000, 2500000
```

I can't make sense of that to be honest. But I wonder if I could just tell it to either lower the threshold or throttle more aggressively?

The crit and high values in sensors look incorrect to me. Is that the problem?

```
~$ sensors
coretemp-isa-0000
Adapter: ISA adapter
Package id 0:  +75.0°C  (high = +100.0°C, crit = +100.0°C)
Core 0:        +74.0°C  (high = +100.0°C, crit = +100.0°C)
Core 1:        +75.0°C  (high = +100.0°C, crit = +100.0°C)

iwlwifi_1-virtual-0
Adapter: Virtual device
temp1:        +65.0°C  

BAT0-acpi-0
Adapter: ACPI interface
in0:           8.42 V  
curr1:            N/A  

dell_smm-virtual-0
Adapter: Virtual device
CPU:          +84.0°C  
Other:        +55.0°C  
Other:        +47.0°C  
Other:        +51.0°C  
Ambient:      +62.0°C  
SODIMM:       +56.0°C  

pch_skylake-virtual-0
Adapter: Virtual device
temp1:        +65.5°C  

nvme-pci-3a00
Adapter: PCI adapter
Composite:    +42.9°C  (low  = -20.1°C, high = +77.8°C)
                       (crit = +81.8°C)
Sensor 1:     +42.9°C  (low  = -273.1°C, high = +65261.8°C)

acpitz-acpi-0
Adapter: ACPI interface
temp1:        +27.8°C  (crit = +119.0°C)
temp2:        +29.8°C  (crit = +119.0°C)
```
Those sysfs nodes have no write access. I suppose I'm not supposed to be able to adjust those levels? Am I holding it wrong?

**Comment by spandruvada** [comment]
What is the kernel version you are using?
Not sure if you have this fix
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/drivers/thermal/intel/int340x_thermal?h=v5.16-rc3&id=8b4bd256674720709a9d858a219fcac6f2f253b5

**Comment by tbaumann** [comment]
> What is the kernel version you are using? Not sure if you have this fix https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/drivers/thermal/intel/int340x_thermal?h=v5.16-rc3&id=8b4bd256674720709a9d858a219fcac6f2f253b5

Oh thanks a lot. That looks pretty much right on the money.

I doubt I have that fix, Ubuntu 21.10 kernel 5.13.0-22-generic. I will test it and raise it to Ubuntu if it's the fix.

**Comment by tbaumann** [comment]
> What is the kernel version you are using? Not sure if you have this fix https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/drivers/thermal/intel/int340x_thermal?h=v5.16-rc3&id=8b4bd256674720709a9d858a219fcac6f2f253b5

Oh thanks a lot. That looks pretty much right on the money.

I doubt I have that fix, Ubuntu 21.10 kernel 5.13.0-22-generic. I will test it and raise it to Ubuntu if it's the fix.

PS: Sorry for the late reply. My reply fell between the cracks.

**Comment by tbaumann** [comment]
That fix went into the latest Ubuntu kernel on  2021-11-03 (LP: #1949636) before I could reasonably test it.

And I haven't seen any shutdowns for a while. I think @spandruvada you were absolutely spot on. Thanks for your help.

Closing and thanks
